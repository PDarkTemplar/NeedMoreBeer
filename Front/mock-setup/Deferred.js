// @flow
class Deferred {
    resolve: (value?: any) => void;
    reject: (reason?: any) => void;
    promise: Promise<any>;
    constructor() {
        this.promise = new Promise((resolve, reject) => {
            this.resolve = resolve;
            this.reject = reject;
        });
    }
}

export default Deferred;
