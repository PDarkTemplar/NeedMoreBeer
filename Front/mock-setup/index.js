// @flow

jest.mock('~/services/i18n', () => ({ get: () => ({ t: key => key }) }));
jest.mock('~/utils', () => ({ generateUUID: () => 'test-uuid' }));

window.BASE_NAME = '/';
window.REST_UI_URL = 'test';
