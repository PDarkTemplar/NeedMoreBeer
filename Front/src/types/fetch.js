// @flow
import { AxiosXHRConfigBase } from 'axios';
import { IObservableValue } from 'mobx';

export type RequestStatus = 'pending' | 'fulfilled' | 'rejected' | 'canceled';

export interface FetchResponseType<T> {
    status: IObservableValue<RequestStatus>;
    requestId: number;
    promise: Promise<T>;
};

export interface AxiosConfig extends AxiosXHRConfigBase<any> {
    data?: any;
    method?: string;
    url?: string;
    reThrow?: boolean;
    withoutAutoAbort?: boolean;
}
