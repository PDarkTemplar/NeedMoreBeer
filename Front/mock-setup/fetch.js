// @flow
import constants from '~/constants';

import Deferred from './Deferred';

type UrlResult = {
    [string]: any,
};

export default (value: any, status: string = constants.fetch.status.fulfilled) => ({
    get: () => ({
        status: { get: () => status },
        // $FlowFixMe
        promise: new Promise(resolve => {
            resolve(value);
        }),
    }),
});

export function resultByUrl(map: UrlResult, status: string = constants.fetch.status.fulfilled) {
    return {
        get: (url: string) => ({
            status: { get: () => status },
            // $FlowFixMe
            promise: new Promise(resolve => {
                resolve(map[url]);
            }),
        }),
    };
}

export function fetchResult(value: any, status: string = constants.fetch.status.fulfilled) {
    return {
        status: { get: () => status },
        // $FlowFixMe
        promise: new Promise(resolve => {
            resolve(value);
        }),
    };
}

export function deferredFetchResult(
    deferred: Deferred,
    requestId?: number,
    status: string = constants.fetch.status.fulfilled
) {
    return {
        status: { get: () => status },
        // $FlowFixMe
        promise: deferred.promise,
        requestId,
    };
}

export function flushPromises() {
    // $FlowFixMe
    return new Promise(resolve => setTimeout(resolve, 0));
}
