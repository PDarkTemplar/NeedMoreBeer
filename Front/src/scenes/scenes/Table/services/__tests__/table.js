// @flow
import 'mock-setup';
import Deferred from 'mock-setup/Deferred';
import { deferredFetchResult } from 'mock-setup/fetch';

import TableView from '../../viewModels/Table';
import TableService from '../Table';

test('load next page', async () => {
    const tableView = new TableView();

    const tableResultD = new Deferred();

    const s: any = {
        viewModels: {
            table: tableView,
        },
        services: {
            apiClient: {
                tableGet: () => deferredFetchResult(tableResultD),
            },
        },
    };

    const service = new TableService(s);

    tableResultD.resolve({
        data: [{ id: 1 }, { id: 2 }],
        numberOfPages: 1559,
        totalResults: 77930,
    });

    await service.loadNextPage();

    expect(tableView).toMatchSnapshot();
});
