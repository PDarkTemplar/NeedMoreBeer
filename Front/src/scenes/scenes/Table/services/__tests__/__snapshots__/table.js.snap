// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load next page 1`] = `
Table {
  "currentPage": 1,
  "filter": Filter {
    "filters": Object {
      "abv": Object {
        "name": "table:abv",
        "value": "",
      },
      "ibu": Object {
        "name": "table:ibu",
        "value": "",
      },
      "name": Object {
        "name": "table:name",
        "value": "",
      },
    },
  },
  "loadedItems": Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
  ],
  "loading": false,
  "numberOfPage": 1559,
  "order": Order {
    "order": "name",
    "sort": "asc",
  },
  "totalRows": 77930,
}
`;
